hail/src/main/scala/is/hail/expr/ir/Requiredness.scala

```diff
@@ -522,6 +529,8 @@ class Requiredness(val usesAndDefs: UsesAndDefs, ctx: ExecuteContext) {
         requiredness.union(aReq.required && stopReq && lookup(start).required && lookup(step).required)
       case ArraySort(a, l, r, c) =>
         requiredness.unionFrom(lookup(a))
+      case ArrayMaximalIndependentSet(a, _) =>
+        requiredness.union(lookup(a).required)
```

Comment:
Why would this screw up the requiredness of the element type? I never touch the element type, why would the requiredness change?

Reply:
Every node in an IR gets a nested RType, and each requiredness rule needs to loosen that rtype for any piece of the nested type that can be optional. Here, you're loosening the top-level requiredness bit but nothing internal. Look up the rule for ArraySort -- that also doesn't change the element requiredness, but it still needs to call `unionFrom(requiredness of stream child)` so that the requiredness of the sort matches its input. 

You can't use `unionFrom` here since the types don't match exactly, you'll need to look up the requiredness of the first struct field of the input and union that with the node's requiredness element.